Copyedit "HTML Attributes" documentation for clarity

hughlilly opened the pull request:
### Description

Copyedit of the HTML Attributes documentation; also fixes a couple of typos.

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=465) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=465) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=465) it.</sub>

yangc95 commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -1,6 +1,6 @@
 ---
 Title: 'Attributes'
-Description: 'HTML attributes can be added to the opening tag of an HTML element to change the elements default behavior or provide additional information about the element. They consist of a name and a value using the following syntax: name="value".'
+Description: 'Attributes can be added to the opening tag of an HTML element to change its default behavior or provide additional data about it. Attributes consist of a name and a value in the format name="value".'
```
```suggestion
Description: 'Attributes can be added to the opening tag of an HTML element to change its default behavior or provide additional data about it.'
```

yangc95 commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -30,29 +30,27 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
 - `codecademy_logo.png` is the attribute value.
 
-All HTML elements can have attributes. Multiple of them, too!
+All HTML elements can have attributes; many have more than one.
```
```suggestion
All HTML elements can have attributes and many have more than one.
```

hughlilly replied:
I think two clauses, as I had it, is better here. At the very least I would suggest a compromise with a comma before the "and":

> All HTML elements can have attributes, and many have more than one.

yangc95 replied:
Looking over this, I'm good with your initial change if we want to separate the points. Otherwise, I would suggest "All HTML elements can have one or more attributes" or just "HTML elements can have one or more attributes."  Lmk what you 🧠  think!

I'm also seeing now that the following code snippet is not prefaced so I'm going to suggest that we add on "The example below displays an image element with three attributes:"

hughlilly replied:
Done!

yangc95 commented on content/html/concepts/attributes/attributes.md:
````diff
@@ -30,29 +30,27 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
 ```
 
-**Note:** Attribute names are case insensitive, but it is a good practice to keep HTML markup lowercase.
+**Note:** Attribute names are not case-sensitive, but it is good practise to write HTML markup in lowercase.
````
```suggestion
**Note:** Attribute names are not case-sensitive, but it is good practice to write HTML markup in lowercase.
```

yangc95 commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -30,29 +30,27 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
 ## Unique ID Attributes
 
-In HTML, specific and unique `id` attributes can be assigned to different elements in order to differentiate between them.
+Specific and unique `id` attributes can be assigned to elements to differentiate them, and `id` values can be referenced in CSS and JavaScript to manipulate, format, and perform specific instructions on a single element alone, without affecting others of the same type.
```
```suggestion
`id` attributes are specific and unique. They can be assigned to differentiate elements. They can also be referenced in CSS and JavaScript to manipulate, format, and perform specific instructions on a single element without affecting similar tags.
```

hughlilly replied:
Maybe we could be even clearer/more succinct here?

> The `id` attribute is generally used to differentiate elements. It is commonly referenced in CSS and JavaScript to manipulate, format, and perform specific instructions on a single element without affecting other elements of the same tag. Its value should be unique.

yangc95 replied:
I think it's more so that the `id` attribute is used to identify elements. Other than that, loving that explanation. Let's change it to what you wrote here.

hughlilly replied:
Awesome, thanks!

yangc95 commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -30,29 +30,27 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
-When needed, the `id` value can be called upon by CSS and JavaScript to manipulate, format, and perform specific instructions on that element and that element only.
-
-Valid `id` attributes should begin with a letter and should only contain letters (`a`-`Z`), digits (`0`-`9`), hyphens (`-`), underscores (`_`), and periods (`.`).
+Valid `id` attribute values begin with a letter and must comprise only letters (`a`-`Z`), digits (`0`-`9`), hyphens (`-`), underscores (`_`), and periods (`.`).
```
```suggestion
Valid `id` attribute values begin with a letter and must be comprised of only letters (`a`-`Z`), digits (`0`-`9`), hyphens (`-`), underscores (`_`), and periods (`.`).
```

hughlilly replied:
Sorry, I don't agree that "comprised of" is grammatically correct. "Must comprise" reads fine to me.

yangc95 replied:
Whoops, I meant to write "composed of" as per https://www.merriam-webster.com/words-at-play/can-you-use-comprised-of-grammar. Thanks for pointing out my error here 👍 We'll just keep this as "must comprise only." Go ahead and resolve this convo

yangc95 commented on content/html/concepts/attributes/attributes.md:
````diff
@@ -30,29 +30,27 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
 ```html
 <h1 id="th1s_is.v4lid">Hello World</h1>
 ```
 
````
Would be helpful to add a note here about what is convention.

hughlilly replied:
Sorry, could you expand what you mean by that? I'm not sure I understand.

yangc95 replied:
Yup! I'm speaking about id naming conventions. What do you think about adding a note that while an id like "th1s_is_v4lid" is valid, it's best to stick to one style? ..camelCase, dash-case, snake_case, and so on.

This is definitely ❓ optional

hughlilly replied:
Added at https://github.com/hughlilly/docs/commit/6e0f64a85ce8792bac8e9da6d578f48a336661a0

yangc95 commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -30,29 +30,27 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
 ## Example
 
-In the provided example, we are giving the `<p>` paragraph element an unique identifier using the `id` attribute and changing the color of the default text using the `style` attribute:
+In this example, the `<p>` (paragraph) element is given a unique identifier in its `id` attribute, and the text color is set using the `style` attribute:
```
```suggestion
In the example below, the `<p>` (paragraph) element is given a unique identifier in its `id` attribute, and the text color is set using the `style` attribute:
```

yangc95 reviewed (commented):
@hughlilly 😸 some nice catches.

First review underway 1️⃣ 
Please take a look at my suggestions/comments and make sure to tag me once you've resolved everything. Let me know if you have any questions.

yangc95 reviewed (approved):
This one is ready to roll 🌯  for review 2️⃣ @KTom101 or @Dusch4593!

hughlilly commented:
I've just added another commit to this. I noticed the `format:verify` test was failing, so I read [CONTRIBUTING.md](CONTRIBUTING.md) for help on how to run the formatter locally.

Here's what I got when I ran this command:

`yarn format content/html/concepts/attributes/attributes.md`

```bash
yarn run v1.22.17
$ prettier --ignore-path .prettierignore --write content/html/concepts/attributes/attributes.md
/bin/sh: prettier: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

This obviously didn't work! Prettier wasn't installed. So I ran `yarn install`, which gave this output:

```bash
yarn install v1.22.17
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > @codecademy/eslint-config@3.1.7" has incorrect peer dependency "eslint@^6.5.1".
warning "@codecademy/eslint-config > eslint-plugin-react@7.16.0" has incorrect peer dependency "eslint@^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0".
[4/4] 🔨  Building fresh packages...
✨  Done in 27.43s.
```

I didn't see a note on running "yarn install" in the contributing file or  in [the "How To…" post](https://dev.to/codecademy/how-to-contribute-to-codecademy-docs-1a77). (Was this the right thing to run? If so, should the documentation be updated?)

In any case, it seems that I can now run the "yarn format" command:

```bash
yarn run v1.22.17
$ prettier --ignore-path .prettierignore --write ./content/html/concepts/attributes/attributes.md
content/html/concepts/attributes/attributes.md 112ms
✨  Done in 1.12s.
```

So the new commit I just added fixes a whitespace issue that apparently existed in `attributes.md`.

Are there other tests I can/should run? Should contributors run these often, or is it OK to let them be run in the GH Action?

Many thanks,
Hugh

Dusch4593 commented:
> Are there other tests I can/should run? Should contributors run these often, or is it OK to let them be run in the GH Action?

@hughlilly Great question! The formatting steps in `CONTRIBUTING.md` are actually [being worked into a hook](https://github.com/Codecademy/docs/pull/265) (when you stage your changes for commit) that runs Prettier/lint on only the files that were staged (ideally, just the files you worked on). This will hopefully lift the burden of formatting/linting from the workload of you, the contributors, and us, the maintainers.

For now, the steps you took to address the whitespace issue are precisely the ones you should take when the `format` test fails. Thank you for doing that! 😄

hughlilly commented:
> The formatting steps in `CONTRIBUTING.md` are actually [being worked into a hook](https://github.com/Codecademy/docs/pull/265) (when you stage your changes for commit) that runs Prettier/lint on only the files that were staged (ideally, just the files you worked on). This will hopefully lift the burden of formatting/linting from the workload of you, the contributors, and us, the maintainers.

Ah, thanks; I’d seen and read through that PR but am not yet familiar enough with development to work out exactly what was being discussed. It’s great that it’s being worked on; sounds like it will make everything easier for all parties.

Dusch4593 commented:
> > The formatting steps in `CONTRIBUTING.md` are actually [being worked into a hook](https://github.com/Codecademy/docs/pull/265) (when you stage your changes for commit) that runs Prettier/lint on only the files that were staged (ideally, just the files you worked on). This will hopefully lift the burden of formatting/linting from the workload of you, the contributors, and us, the maintainers.
> 
> Ah, thanks; I’d seen and read through that PR but am not yet familiar enough with development to work out exactly what was being discussed. It’s great that it’s being worked on; sounds like it will make everything easier for all parties.

That's the hope! 😃

SSwiniarski commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -30,29 +30,33 @@ So for example, the `<img>` image element has a mandatory `src` attribute that c
 - `codecademy_logo.png` is the attribute value.
 
-All HTML elements can have attributes. Multiple of them, too!
+HTML elements can have one or more attributes.
```
```suggestion
HTML elements can have any number of attributes.
```

They can have zero attributes as well.

SSwiniarski commented on content/html/concepts/attributes/attributes.md:
```diff
@@ -13,9 +13,9 @@ CatalogContent:
   - 'paths/front-end-engineer-career-path'
```
Fix the spacing above for Tags and 'learn-html'

hughlilly replied:
This appears to be fixed now, but I'm confused as to why it appears this way in the code review. If there's something I haven't done right, please let me know. Otherwise I think everything is as it should be. Thanks.

SSwiniarski reviewed (changes requested):
I just did a 2nd review @hughlilly, resolve the edits and things should be good to go.

hughlilly commented:
> I just did a 2nd review @hughlilly, resolve the edits and things should be good to go.

I believe everything has been fixed up. Regarding your comment on the layout of the front matter (is that what it's called), I believe this was fixed in the commit after the one you reviewed, but do let me know if this isn't the case.

Thanks
Hugh

yangc95 commented:
🎁  @hughlilly your entry is live here!

https://www.codecademy.com/resources/docs/html/attributes

cc: @SSwiniarski @Dusch4593
